libsolidity/formal/SMTEncoder.cpp

```diff
@@ -1296,8 +1296,27 @@ void SMTEncoder::addArrayLiteralAssertions(
 )
 {
 	m_context.addAssertion(_symArray.length() == _elementValues.size());
+
+	// Assert to the solver that _elementValues are exactly the elements at the beginning of the array.
+	// Since we create new symbolic representation for every array literal in the source file, we want to ensure that
+	// representations of two equal literals are also equal. For this reason we always start from constant-zero array.
+	// This ensures SMT-LIB arrays (which are infinite) are also equal beyond the length of the Solidity array literal.
+	auto type = _symArray.type();
+	smtAssert(type);
+	auto valueType = [&]() {
```

Comment:
You can leave it like it is but I think that first declaring a var
and then using if statements to assign to it is more readable.

Reply:
I actually disagree with that, splitting the declaration and initialization is often considered an anti-pattern.
See for example Core Guidelines [ES 22](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es22-dont-declare-a-variable-until-you-have-a-value-to-initialize-it-with), [ES 28](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es28-use-lambdas-for-complex-initialization-especially-of-const-variables), or [this talk](https://www.youtube.com/watch?v=VV9vwFsaQ6U).

Reply:
Right. I never thought about this like that. Makes sense.